components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -98,13 +104,15 @@ public Secret getSecret(String secretName) throws SecretManagementException {
     }
 
     @Override
-    public Secrets getSecrets() throws SecretManagementException {
+    public Secrets getSecrets(String secretTypeName) throws SecretManagementException {
 
         validateSecretManagerEnabled();
-        List secretList = this.getSecretDAO().getSecrets(getTenantId());
+        validateSecretsRetrieveRequest(secretTypeName);
+        SecretType secretType = getSecretType(secretTypeName);
+        List secretList = this.getSecretDAO().getSecrets(secretType.getId(), secretType.getName(), getTenantId());
```

Comment:
why we need to pass secretType and name ?
If we want both, better to pass the SecretType object itself

Reply:
This is name of the secret Type and name(alias) of the secret.
eg- secret type: adaptive
      secretname : choreoSampleService1

Reply:
resolved